Solutions/Branwen/GUI.cs

```diff
@@ -0,0 +1,166 @@
+﻿using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Spreadsheet;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+
+namespace Branwen
+{
+    public partial class GUI : Form
+    {
+        private int fileCount = 0;
+		
+        public GUI()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Button Handler, Selects Directory then does Inventory
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void buttonSelectAndRunInventory_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                //make this thing and all associated stuff go away
+                buttonSelectAndRunInventory.Enabled = false;
+                buttonSelectAndRunInventory.Text = "Working";
+                FolderBrowserDialog folderDialog = new FolderBrowserDialog();
+                folderDialog.Description = "Set Folder to Inventory";
+                if (folderDialog.ShowDialog() != DialogResult.OK)
+                {
+                    return;
+                }
+
+                DirectoryInfo[] topLevelDirectories = new DirectoryInfo(folderDialog.SelectedPath).GetDirectories();
+                string outputFile = Path.Combine(folderDialog.SelectedPath, "MediadriveInventory.xlsx");
+                if (topLevelDirectories == null || outputFile == null)
+                {
+                    MessageBox.Show("This Directory is Invalid. Please try again");
+                    return;
+                }
+
+                if(File.Exists(outputFile))
+                {
+                    File.Delete(outputFile);
+                }
+
+                //Create worksheet
+                SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Create(outputFile, DocumentFormat.OpenXml.SpreadsheetDocumentType.Workbook);
+                WorkbookPart workbookpart = spreadsheetDocument.AddWorkbookPart();
+                workbookpart.Workbook = new Workbook();
+                Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild<Sheets>(new Sheets());
+
+                //Loop through all Directories and make a new sheet and put that shit in there
+                for (int i = 0; i < topLevelDirectories.Length; i++)
+                {
+                    WorksheetPart worksheetPart = workbookpart.AddNewPart<WorksheetPart>();
+                    worksheetPart.Worksheet = new Worksheet(new SheetData());
+                    SheetData sheetData = worksheetPart.Worksheet.GetFirstChild<SheetData>();
+                    Sheet sheet = new Sheet();
+                    sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
+                    sheet.Name = topLevelDirectories[i].Name;
+                    sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
+                    sheets.Append(sheet);

```

Comment:
does this method take a Sheet object as a parameter?  It looks like it might play more nicely with
-	sheets.Append(new List<OpenXmlElement> { sheet });